assets/recipient-form.js

```diff
@@ -1,142 +1,164 @@
+    onChange() {
+      if (!this.checkboxInput.checked) {
+        this.clearInputFields();
+        this.disableInputFields();
         this.clearErrorMessage();
-        this.errorMessageWrapper.hidden = false;
-        if (typeof body === 'object') {
-          this.errorMessage.innerText = this.defaultErrorHeader;
-          return Object.entries(body).forEach(([key, value]) => {
-            const errorMessageId = `RecipientForm-${key}-error-${this.dataset.sectionId}`;
-            const fieldSelector = `#Recipient-${key}-${this.dataset.sectionId}`;
-            const placeholderElement = this.querySelector(`${fieldSelector}`);
-            const label = placeholderElement?.getAttribute('placeholder') || key;
-            const message = `${label} ${value}`;
-            const errorMessageElement = this.querySelector(`#${errorMessageId}`);
-            const errorTextElement = errorMessageElement?.querySelector('.error-message');
-            if (!errorTextElement) return;
-
-            if (this.errorMessageList) {
-              this.errorMessageList.appendChild(this.createErrorListItem(fieldSelector, message));
-            }
-
-            errorTextElement.innerText = `${message}.`;
-            errorMessageElement.classList.remove('hidden');
-
-            const inputElement = this[`${key}Input`];
-            if (!inputElement) return;
-
-            inputElement.setAttribute('aria-invalid', true);
-            inputElement.setAttribute('aria-describedby', errorMessageId);
-          });
-        }
-
-        this.errorMessage.innerText = body;
+      } else {
+        this.enableInputFields();
       }
+    }
 
-      createErrorListItem(target, message) {
-        const li = document.createElement('li');
-        const a = document.createElement('a');
-        a.setAttribute('href', target);
-        a.innerText = message;
-        li.appendChild(a);
-        li.className = 'error-message';
-        return li;
-      }
+    inputFields() {
+      return [
+        this.emailInput,
+        this.nameInput,
+        this.messageInput,
+        this.sendonInput
+      ];
+    }
 
-      clearErrorMessage() {
-        this.errorMessageWrapper.hidden = true;
+    disableableFields() {
+      return [...this.inputFields(), this.offsetProperty];
+    }
 
-        if (this.errorMessageList) this.errorMessageList.innerHTML = '';
+    clearInputFields() {
+      this.inputFields().forEach((field) => field.value = '');
+    }
 
-        this.querySelectorAll('.recipient-fields .form__message').forEach((field) => {
-          field.classList.add('hidden');
-          const textField = field.querySelector('.error-message');
-          if (textField) textField.innerText = '';
-        });
+    enableInputFields() {
+      this.disableableFields().forEach((field) => field.disabled = false);
+    }
+
+    disableInputFields() {
+      this.disableableFields().forEach((field) => field.disabled = true);
+    }
```

Comment:
Could this be something to add to `this.clearInputFields()` instead of its own function?

I see that `this.clearInputFields()` is run twice and in one instance, `this.disableInputFields()` isn't run, in the `resetRecipientForm()`. Would it make sense to also run it there ?

Reply:
It's different because we never want to clear the offset field but we do want to disable it. It's because offset is not user input but it does sit along side user input and use the same line item property mechanism, so there are two similar use cases but not the same.